### PR TITLE
improve interaction of keepClosingSlash & removeAttributeQuotes

### PR DESCRIPTION
**Pull request by alexlamsl**
I think this will address #414

### CODE REVIEW

**Comment by alexlamsl** [comment]
Actually, hold the horses - I need to think about `attr.customClose`...

**Comment by alexlamsl** [comment]
Okay - it is working as expected even for that corner case; if the attribute is removed by a template engine, the preceding space would still prevent confusion over `/>`

**Comment by alexlamsl** [comment]
@kangax would you mind having a look at this? The code does the following:
- fixes two corner cases as added to the test
- implement removal of quotes around final attribute value if not clashing with closing slash

Just want to get this out of the way before pondering on #2, #415 & #438 as they would involve modifying similar range of code...

**Comment by alexlamsl** [comment]
I guess it's probably easier if I do them all in one PR...

**Comment by alexlamsl** [comment]
I am currently using these modifications on the test server of my web site:
- uses AngularJS & Bootstrap
- minifies static HTML files as well as dynamic templates from REST end-points

And they seem to work well AFAICT.

Should fix #2, #415, #438, #470

File: tests/minifier.js [hunk 1]
```diff
@@ -1293,25 +1315,25 @@
...
 
     equal(minify('<a title="x"href=" ">foo</a>', options), '<a title="x" href="">foo\n</a>');
-    equal(minify('<p id=""class=""title="">x', options), '<p id="" class=""\n title="">x</p>');
+    equal(minify('<p id=""class=""title="">x', options), '<p id="" class="" \ntitle="">x</p>');
     equal(minify('<p x="x\'"">x</p>', options), '<p x="x\'">x</p>', 'trailing quote should be ignored');
     equal(minify('<a href="#"><p>Click me</p></a>', options), '<a href="#"><p>Click me\n</p></a>');
     equal(minify('<span><button>Hit me</button></span>', options), '<span><button>Hit me\n</button></span>');
     equal(minify('<object type="image/svg+xml" data="image.svg"><div>[fallback image]</div></object>', options),
-      '<object\n type="image/svg+xml"\n data="image.svg"><div>\n[fallback image]</div>\n</object>'
+      '<object \ntype="image/svg+xml" \ndata="image.svg"><div>\n[fallback image]</div>\n</object>'
```
Comment (kangax) [inline tests/minifier.js:1323]: Why is there \n here (and in other places)?

Comment (kangax) [inline tests/minifier.js:1323]: Oh, we already had this (max line length). Nevermind.

**Comment by kangax** [comment]
So this actually looks great to me! Thanks for tackling it, @alexlamsl 

@XhmikosR @duncanbeevers any thoughts before I merge this?

**Comment by alexlamsl** [comment]
Added `minimize` to `devDependencies` as per https://github.com/kangax/html-minifier/commit/c4c0bac9cc9a0bb5c7da9646b24b784949352a4f#commitcomment-15621289

Comparison of `node benchmark` between `master` and this PR:

| File | Original | `master` | #468 |
| --- | --- | --- | --- |
| abc | 93187 (91.00 KB) | 77848 (76.02 KB) | 77760 (75.94 KB) |
| amazon | 251828 (245.93 KB) | 216837 (211.75 KB) | 216364 (211.29 KB) |
| eloquentjavascript | 890398 (869.53 KB) | 850539 (830.60 KB) | 850510 (830.58 KB) |
| es6 | 3765813 (3677.55 KB) | 3132089 (3058.68 KB) | 3132080 (3058.67 KB) |
| es6-table | 120768 (117.94 KB) | 82043 (80.12 KB) | 82040 (80.12 KB) |
| google | 136320 (133.13 KB) | 131453 (128.37 KB) | 131132 (128.06 KB) |
| html-minifier | 49978 (48.81 KB) | 38609 (37.70 KB) | 38387 (37.49 KB) |
| msn | 160454 (156.69 KB) | 136465 (133.27 KB) | 136113 (132.92 KB) |
| newyorktimes | 134565 (131.41 KB) | 113790 (111.12 KB) | 113291 (110.64 KB) |
| stackoverflow | 205280 (200.47 KB) | 163534 (159.70 KB) | 161975 (158.18 KB) |
| wikipedia | 411051 (401.42 KB) | 390080 (380.94 KB) | 389515 (380.39 KB) |

**Comment by alexlamsl** [comment]
@kangax reading the repo some more - do I need to commit those files under `/dist`?

They are just auto-generated by `grunt`, but are they getting used directly by [your website](http://kangax.github.io/html-minifier/) from `master`?

**Comment by kangax** [comment]
Yep, it's coming straight from dist/...min.js

**Comment by alexlamsl** [comment]
ok - `dist` files committed

**Comment by kangax** [comment]
Ok, thanks. I'll wait for couple days for @XhmikosR @duncanbeevers

**Comment by XhmikosR** [comment]
I just had a quick look and I have to say great job @alexlamsl!

Since tests are passing, I'd say it's safe to merge and wait a little bit for feedback before releasing a new version @kangax.

**Comment by kangax** [comment]
I'm wondering if this should be 1.2.0 (new feature) or 2.0 (breaking change)

**Comment by XhmikosR** [comment]
I'd go with 1.2.0. It's a new feature bit not a breaking change.

**Comment by genintho** [comment]
@kangax should the different issues mentioned in the PR being closed now that this is merged?

**Comment by kangax** [comment]
Oh yes, thanks!
